src/main/resources/com/google/api/codegen/java/http_test.snip

```diff
@@ -90,6 +90,46 @@
   }
 @end
 
+
+@private longRunningMethodTestCase(test)
+  @@Test
+  @@SuppressWarnings("all")
+  public void {@test.name}() throws Exception {
+    {@initCode(test.mockResponse.rpcResponseInitCode)}
+    Operation resultOperation =
+        Operation.newBuilder()
+            .setName("{@test.name}")
+            .setStatus("DONE")
+            .build();
+    mockService.addResponse(resultOperation);
+
+    {@initCode(test.testCaseInitCode)}
+
+    {@longRunningMethodCall(test)}
+
+    {@unarySuccessAsserts(test)}
+  }
+
+  @@Test
+  @@SuppressWarnings("all")
+  public void {@test.nameWithException}() throws Exception {
+    {@addException(test)}
+
+    try {
+      {@initCode(test.testCaseInitCode)}
+
+      client.{@test.clientMethodName}(\
+          {@sampleMethodCallArgList(test.testCaseInitCode.fieldSettings)}).get();
+      Assert.fail("No exception raised");
+    } catch (ExecutionException e) {
+      Assert.assertEquals(InvalidArgumentException.class, e.getCause().getClass());
```

Comment:
Maybe
```java
Assert.assertTrue(e.getCause() instanceof InvalidArgumentException);
```
is cleaner, and also allows subclasses of `InvalidArgumentException` as valid cause. This seems lika a desired behavior. In real world `catch(InvalidArgumentException e)` will also catch a subclass of `InvalidArgumentException` which is idiomatic for java exceptions, so the tests should treat `InvalidArgumentException` and its potential subclasses as equally correct.